src/emcpy/stats.py

```diff
@@ -285,3 +286,45 @@ def bootstrap(insample, level=.95, estimator='mean', nrepl=10000):
     ci_upper = _np.percentile(deltas, upper_pctile)
 
     return ci_lower, ci_upper
+
+
+def spectrum_stats(df):
+    '''
+    Function to calculate major statistics for radiance data as a function of channel.  
+    Statisics are for observations that pass QC only.
+
+    Args:
+        df : (Pandas dataframe) A data frame containing raw radiance information as produced
+             by the get_data method of Radiance.
+        
+    Returns:
+        channel_stats : (Pandas dataframe) A data frame containing channel-by-channel statistics
+    '''
+
+    print('Calling Spectrum_Stats')
+
+    # Get channel numbers in dataframe
+    sc=df.index.unique(level="Channel")
+
+    # Initialise output dataframe
+    channel_stats = _pd.DataFrame(index=sc,
+        columns = ["count","omf_unadjusted_mean", "omf_adjusted_mean",
+            "omf_unadjusted_stddev", "omf_adjusted_stddev"])
+
+    # Sorting by channel increases efficiency and prevents a warning message
+    dfs=df.sort_index()
+
+    # Loop through channels and calculate stats
+    for chan in sc:
+        # Subset based on channel and QC indexes
+        tmp = dfs.loc[(chan,0.0)]
+        # This next line makes sure that all rejected obs are avoided.
+        # Using the inverse observation error is more reliable than the QC flag.
+        tmp = tmp[(tmp['inverse_observation_error'] > 0.0)]
+        count = tmp.size
+        channel_stats["count"].loc[chan] = count
```

Comment:
Not a big deal but you could skip the assignment to count and just set channel_stats["count"].loc[chan] = tmp.size.

Reply:
@EdwardSafford-NOAA You are right.  I have made the change.